packages/server/src/workers/scheduledtiming.ts

```diff
@@ -0,0 +1,290 @@
+import { createReference, Operator } from '@medplum/core';
+import { AuditEvent, Bot, Practitioner, ProjectMembership, Reference, Resource, Timing } from '@medplum/fhirtypes';
+import { Job, Queue, QueueBaseOptions, Worker } from 'bullmq';
+import { MedplumRedisConfig } from '../config';
+import { executeBot } from '../fhir/operations/execute';
+import { systemRepo } from '../fhir/repo';
+import { logger } from '../logger';
+import { AuditEventOutcome } from '../util/auditevent';
+import { isValidCron } from 'cron-validator';
+
+const daysOfWeekConversion = { sun: 0, mon: 1, tue: 2, wed: 3, thu: 4, fri: 5, sat: 6 };
+
+/*
+ * The ScheduleTiming worker inspects resources takes a bot,
+ * if it has the scheduledTiming property, will add it as a repeatable
+ * Cron job
+ */
+// Repeatable is based on BullMQ docs https://docs.bullmq.io/guide/jobs/repeatable
+interface Repeatable {
+  repeat: {
+    pattern?: string;
+    every?: number;
+    limit?: number;
+  };
+  jobId?: string;
+}
+
+export interface ScheduledTimingJobData {
+  readonly resourceType: string;
+  readonly id: string;
+}
+
+const queueName = 'ScheduledTimingQueue';
+const jobName = 'ScheduledTimingJobData';
+
+let queue: Queue<ScheduledTimingJobData> | undefined = undefined;
+let worker: Worker<ScheduledTimingJobData> | undefined = undefined;
+
+/**
+ * Initializes the scheduled timing worker.
+ * Sets up the BullMQ job queue.
+ * Sets up the BullMQ worker.
+ */
+export function initScheduledTimingWorker(config: MedplumRedisConfig): void {
+  const defaultOptions: QueueBaseOptions = {
+    connection: config,
+  };
+
+  queue = new Queue<ScheduledTimingJobData>(queueName, {
+    ...defaultOptions,
+    defaultJobOptions: {
+      attempts: 3,
```

Comment:
Oy, retry policy.  This seems like a very sensible default, and a good place to start 👍 

I filed a separate ticket to track this: https://github.com/medplum/medplum/issues/1749